### ISSUE TITLE
[Bug] Audioonly model not learning

### ISSUE THREAD

**Issue by rlleshi** [open-mmlab/mmaction2#2523]
### Branch

main branch (1.x version, such as `v1.0.0`, or `dev-1.x` branch)

### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmaction2/issues) and [Discussions](https://github.com/open-mmlab/mmaction2/discussions) but cannot get the expected help.
- [X] I have read the [documentation](https://mmaction2.readthedocs.io/en/latest/) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmaction2).

### Environment

```
sys.platform: linux
Python: 3.9.0 (default, Jul 27 2022, 17:09:22) [GCC 8.4.0]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0: NVIDIA GeForce RTX 2070 Super with Max-Q Design
CUDA_HOME: /usr
NVCC: Cuda compilation tools, release 10.1, V10.1.24
GCC: gcc (Ubuntu 8.4.0-3ubuntu2) 8.4.0
PyTorch: 1.10.0+cu113
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.2
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.11.1+cu113
OpenCV: 4.7.0
MMEngine: 0.7.2
MMAction2: 1.0.0+b7af9d3
```

### Describe the bug

I've been trying to train an audio recognition model using either `ResNetAudio` or plain `ResNet` as backbones. But the model isn't learning at all. The training accuracy keeps looping around different random values below 0.5 (that don't make much sense), or in another scenario, they were just alternating between 0.0 & 1.0, which, again, makes no sense.

The validation accuracy on the other hand never changes from the first epoch, onwards. 

I have tried various settings such as batch sizes, learning rates, and clip_lens. But it doesn't really help (a quite high learning rate was responsible for the second scenario of 0.0 & 1.0 mentioned above).

As a note, the training with the same models and dataset works as expected with the previous main branch of mmaction2 (what is now `0.X`). 

The implementation of the backbones or head hasn't changed at this version though, so I am at a loss about this. Any ideas?

P.S.
This issue seems to pertain to STGCN as well: #2496.

**Comment by Dai-Wenxun** [comment]
I upgrade this part from 0.x to 1.x. In my memory, I just fix a little bug in `MelSpectrogram` at [here](https://github.com/open-mmlab/mmaction2/blob/main/mmaction/datasets/transforms/processing.py#LL1374C74-L1374C74) compared to [0.x](https://github.com/open-mmlab/mmaction2/blob/0.x/mmaction/datasets/pipelines/augmentations.py#LL1893C36-L1893C36). I did not train this model for the loss of the data, so I'm also confused about your issue.

**Comment by rlleshi** [comment]
@Dai-Wenxun I tested the model trained with mmaction2 `0.x` with a config from `1.x` and the results are gibberish (`acc/mean1: 0.2500` for 4 classes).

Actually, it's the same gibberish result that's coming as validation accuracy when training with `1.x`. Ofc the accuracy of the model is calculated accordingly in `0.x`.

**Comment by Dai-Wenxun** [comment]
We will figure it out this week.

**Comment by Dai-Wenxun** [comment]
Could you please show me your full config?

**Comment by rlleshi** [comment]
Sure @Dai-Wenxun 

``` 
_base_ = ['../_base_/default_runtime.py']

# model settings
num_classes = 4
model = dict(
    type='RecognizerAudio',
    backbone=dict(
        # type='ResNetAudio',
        # depth=50,
        type='ResNet',
        depth=18,
        pretrained=None,
        in_channels=1,
        norm_eval=False),
    cls_head=dict(
        type='TSNAudioHead',
        num_classes=num_classes,
        # in_channels=1024,
        in_channels=512,
        dropout_ratio=0.5,
        init_std=0.01,
        average_clips='prob',
        topk=(1, 2, 3, 4, 5)))

# dataset settings
dataset_type = 'AudioDataset'
data_root = ...
data_root_val = ...
ann_file_train = ...
ann_file_val = ...
ann_file_test = ...

train_pipeline = [
    dict(type='LoadAudioFeature'),
    dict(type='SampleFrames', clip_len=64, frame_interval=2, num_clips=1),
    dict(type='AudioFeatureSelector'),
    dict(type='FormatAudioShape', input_format='NCTF'),
    dict(type='PackActionInputs')
]
val_pipeline = [
    dict(type='LoadAudioFeature'),
    dict(
        type='SampleFrames',
        clip_len=64,
        frame_interval=2,
        num_clips=1,
        test_mode=True),
    dict(type='AudioFeatureSelector'),
    dict(type='FormatAudioShape', input_format='NCTF'),
    dict(type='PackActionInputs')
]
test_pipeline = [
    dict(type='LoadAudioFeature'),
    dict(
        type='SampleFrames',
        clip_len=64,
        frame_interval=2,
        num_clips=10,
        test_mode=True),
    dict(type='AudioFeatureSelector'),
    dict(type='FormatAudioShape', input_format='NCTF'),
    dict(type='PackActionInputs')
]

train_dataloader = dict(
    batch_size=32,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        ann_file=ann_file_train,
        data_prefix=dict(audio=data_root),
        suffix='.npy',
        pipeline=train_pipeline))
val_dataloader = dict(
    batch_size=32,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        ann_file=ann_file_val,
        pipeline=val_pipeline,
        data_prefix=dict(audio=data_root_val),
        suffix='.npy',
        test_mode=True))
test_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        ann_file=ann_file_test,
        pipeline=test_pipeline,
        data_prefix=dict(audio=data_root_val),
        suffix='.npy',
        test_mode=True))

val_evaluator = dict(type='AccMetric',
                     metric_options=dict(top_k_accuracy=dict(topk=(1, 2, 3, 4, 5))))
test_evaluator = dict(type='AccMetric',
                     metric_options=dict(top_k_accuracy=dict(topk=(1, 2, 3, 4, 5))))

train_cfg = dict(
    type='EpochBasedTrainLoop', max_epochs=80, val_begin=1, val_interval=1)
val_cfg = dict(type='ValLoop')
test_cfg = dict(type='TestLoop')

param_scheduler = [
    dict(type='CosineAnnealingLR', eta_min=0, T_max=100, by_epoch=True)
]

optim_wrapper = dict(
    optimizer=dict(type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001),
    clip_grad=dict(max_norm=40, norm_type=2))

# runtime settings
default_hooks = dict(
    checkpoint=dict(max_keep_ckpts=3, interval=5), logger=dict(interval=20))

# load_from = 'work_dir/audio/tsn_r18_8xb320-64x1x1-100e_kinetics400-audio-feature_20201012-bf34df6c.pth'

```

**Comment by Dai-Wenxun** [comment]
I used the [cvdf k400 dataset](https://github.com/cvdfoundation/kinetics-dataset#kinetics-400) and audio process scripts in mmaction2 (i.e., [extract_audio.py](https://github.com/open-mmlab/mmaction2/blob/main/tools/data/extract_audio.py) and [build_audio_features.py](https://github.com/open-mmlab/mmaction2/blob/main/tools/data/build_audio_features.py)) to extract audio in the `wav` format and Mel Spectrogram as `npy` files. In total, I got 238,429 training samples and 19,763 validation samples, then I used [tsn_r18_8xb320-64x1x1-100e_kinetics400-audio-feature.py](https://github.com/open-mmlab/mmaction2/blob/bb1b2d867e743077da6be06c814c5f54d6c32fc6/configs/recognition_audio/resnet/tsn_r18_8xb320-64x1x1-100e_kinetics400-audio-feature.py). The [training log](https://github.com/open-mmlab/mmaction2/files/11928896/tsn_r18_8xb320-64x1x1-100e_kinetics400-audio-feature.log) LGTM and the `val_acc` basically aligns with that in [0.x version](https://download.openmmlab.com/mmaction/recognition/audio_recognition/tsn_r18_64x1x1_100e_kinetics400_audio_feature/20201010_144630.log).

All changes can be found in #2570.

If you have a limited number of training samples, your model may struggle to generalize well on the validation set, especially if the audio model lacks pretraining. Therefore, I recommend performing pretraining on the K400 dataset first and then fine-tuning it on your own dataset.

Good luck bro!
